core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -566,14 +565,31 @@ private void runInternal() {
           } finally {
             jumpListener.setListener(listener);
           }
+
+          // An extremely short deadline may expire before the stream listener is set. This causes
```

Comment:
This comment could be seen as misleading. Try to make it clear we are talking about `stream.setListener(jumpListener)` and not `jumpListener.setListener(listener)`